File: backend/shared/src/main/scala/delivery/model/Accounts.scala
package delivery.model

/** 与前端 `frontend/src/delivery/model/accounts.ts` 对齐 */

final case class CustomerAccount(
    role: String,
    username: String,
    password: String,
    profile: CustomerProfile
)

final case class MerchantAccount(
    role: String,
    username: String,
    password: String,
    profile: MerchantProfile
)

final case class RiderAccount(
    role: String,
    username: String,
    password: String,
    profile: RiderProfile
)

final case class AdminAccount(
    role: String,
    username: String,
    password: String,
    displayName: String
)

final case class AccountStore(
    customerAccounts: List[CustomerAccount],
    merchantAccounts: List[MerchantAccount],
    riderAccounts: List[RiderAccount],
    adminAccounts: List[AdminAccount]
)

File: frontend/src/delivery/model/accounts.ts
/** 与后端 `ApiDto` 中公开账号类型及概念上的账户视图对齐（不含 password） */

import type { CustomerProfile, MerchantProfile, RiderProfile } from './profiles'

export interface CustomerAccountPublic {
  role: 'customer'
  username: string
  profile: CustomerProfile
}

export interface MerchantAccountPublic {
  role: 'merchant'
  username: string
  profile: MerchantProfile
}

export interface RiderAccountPublic {
  role: 'rider'
  username: string
  profile: RiderProfile
}

export interface AdminAccountPublic {
  role: 'admin'
  username: string
  displayName: string
}
